corehq/apps/es/case_search.py

```diff
@@ -312,3 +333,69 @@ def flatten_result(hit, include_score=False, is_related_case=False):
     for key in SYSTEM_PROPERTIES:
         result.pop(key, None)
     return result
+
+
+def wrap_case_search_hit(hit, include_score=False, is_related_case=False):
+    """Convert case search index hit to CommCareCaseSQL
+
+    Nearly the opposite of
+    `corehq.pillows.case_search.transform_case_for_elasticsearch`.
+
+    The "case_properties" list of key/value pairs is converted to a dict
+    and assigned to `case_json`. 'Secial' case properties are excluded
+    from `case_json`, even if they were present in the original case's
+    dynamic properties.
+
+    All fields excluding "case_properties" and its contents are assigned
+    as attributes on the case object if `CommCareCaseSQL` has a field
+    with a matching name. Fields like "doc_type" and "@indexed_on" are
+    ignored.
+
+    Warning: `include_score=True` or `is_related_case=True` may cause
+    the relevant user-defined properties to be overwritten.
+
+    :returns: A `CommCareCaseSQL` instance.
+    """
+    from corehq.form_processor.models import CommCareCaseSQL
+    data = hit.get("_source", hit)
+    _SPECIAL_PROPERTIES = SPECIAL_CASE_PROPERTIES_MAP
```

Comment:
what's the purpose of assigning these to local variables? Is there a performance implication?

Reply:
Yes, global variable lookups are more complex than local refs. https://stackoverflow.com/a/12590209/10840